### PR TITLE
feat(exporter/sumologic): drop attribute used by routing processor

### PR DESCRIPTION
**Pull request by sumo-drosiek**
Drop metrics attribute used for routing

### CODE REVIEW

**Comment by sumo-drosiek** [comment]
I'm going to add it for logs and traces, even if we didn't have that case for them yet

**Review by swiatekm-sumo — COMMENTED** [review]
Can you also add a test for traces?

File: pkg/exporter/sumologicexporter/README.md [hunk 1]
```diff
@@ -58,6 +58,13 @@ exporters:
     # source templating is going to be applied,
     # default = `%{_metric_}`
     graphite_template: <graphite_template>
+    # name of resource attribute which should be dropped for records
+    # this is for attribute used by routing processor
+    # other attributes should be removed by processors in pipelines before
+    # This is workaround for the following issue:
+    # https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/7407
+    # default = `sumo_routing`
+    drop_routing_atttribute: <drop_routing_atttribute>
```
Comment (pmalek-sumo) [inline pkg/exporter/sumologicexporter/README.md:67]: The name is a bit unfortunate IMHO. It indicates that it should be a bool.

File: pkg/exporter/sumologicexporter/config.go [hunk 2]
```diff
@@ -255,4 +260,6 @@ const (
 	DefaultTimestampKey string = "timestamp"
 	// DefaultFlattenBody defines default FlattenBody value
 	DefaultFlattenBody bool = false
+	// DefaultDropRoutingAttribute defines default DropRoutingAttribute
+	DefaultDropRoutingAttribute string = "sumo_routing"
```
Comment (pmalek-sumo) [inline pkg/exporter/sumologicexporter/config.go:264]: I wonder if we shouldn't set the default to empty string and don't drop anything then (make it conditional).

@SumoLogic/open-source-collection-team  WDYT?

File: pkg/exporter/sumologicexporter/README.md (hunk 1 above)
Comment (sumo-drosiek) [reply pkg/exporter/sumologicexporter/README.md:67]: would `routing_attribute_to_drop` be a better name?

File: pkg/exporter/sumologicexporter/config.go (hunk 2 above)
Comment (sumo-drosiek) [reply pkg/exporter/sumologicexporter/config.go:264]: I agree, especially if we do not target it by defaut to `http_listener_v2_path`

File: pkg/exporter/sumologicexporter/README.md (hunk 1 above)
Comment (pmalek-sumo) [reply pkg/exporter/sumologicexporter/README.md:67]: Not sure about either of those 😬 . But let's go with your proposal: `routing_attribute_to_drop`

Comment (sumo-drosiek) [reply pkg/exporter/sumologicexporter/README.md]: fixed then :shrug:

File: pkg/exporter/sumologicexporter/config.go (hunk 2 above)
Comment (sumo-drosiek) [reply pkg/exporter/sumologicexporter/config.go]: fixed

**Review by swiatekm-sumo — APPROVED** [review]

**Pull request merged**